types/openapi-factory/index.d.ts

```diff
@@ -0,0 +1,49 @@
+// Type definitions for openapi-factory 4.2
+// Project: https://github.com/wparad/openapi-factory.js
+// Definitions by: Daan Boerlage <https://github.com/runebaas>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+
+declare namespace OpenApi {
+    interface OpenApiOptions {
+        requestMiddleware?: () => any;
+        responseMiddleware?: () => any;
+        errorMiddleware?: () => any;
+    }
+}
+
+declare class OpenApi {
+    constructor(options: OpenApi.OpenApiOptions, overrideLogger?: () => void);
+
+    setAuthorizer(authorizerFunc: (req?: any) => any): void;
+    onEvent(onEventFunc: (req?: any) => any): void;
+    onSchedule(onScheduleFunc: (req?: any) => any): void;
+
+    head(route: string, p0: (req?: any) => any): void;
+    head(route: string, p0: object, p1: (req?: any) => any): void;
+
+    get(route: string, p0: (req?: any) => any): void;
+    get(route: string, p0: object, p1: (req?: any) => any): void;
+
+    post(route: string, p0: (req?: any) => any): void;
+    post(route: string, p0: object, p1: (req?: any) => any): void;
+
+    put(route: string, p0: (req?: any) => any): void;
+    put(route: string, p0: object, p1: (req?: any) => any): void;
+
+    patch(route: string, p0: (req?: any) => any): void;
+    patch(route: string, p0: object, p1: (req?: any) => any): void;
+
+    delete(route: string, p0: (req?: any) => any): void;
+    delete(route: string, p0: object, p1: (req?: any) => any): void;
+
+    options(route: string, p0: (req?: any) => any): void;
+    options(route: string, p0: object, p1: (req?: any) => any): void;
+
+    any(route: string, p0: (req?: any) => any): void;
```

Comment:
* can we call `p1` (`p0` when there is no `p1`) => handler` which is a `(req?: object) => Promise<any>)`

* can we call `p0` (when there is a `p1`) options (when , and include the `Options` object here which contains the `boolean` `rawBody`?)